src/core/memory.d

```diff
@@ -595,14 +595,14 @@ struct GC
      * Returns:
      *  The base address of the memory block referenced by p or null on error.
      */
-    static inout(void)* addrOf( inout(void)* p ) nothrow /* FIXME pure */
+    static inout(void)* addrOf( inout(void)* p ) pure nothrow @nogc
```

Comment:
Any reason why pure should still be omitted?

Reply:
With the current rules for pure, the second call could be elided by the compiler:
```
    void* p = GC.malloc(10000);
    void* q = GC.addrOf(p);
    GC.free(p);
    void* r = GC.addrOf(p);
    assert(r is null);
```

Reply:
So I should keep the `/* FIXME pure */`.